### DISCUSSION TITLE
Pulse width, pattern and jitter for board and glove

### DISCUSSION THREAD

**Discussion by pdbuzzboard** [pdbuzzboard/pdbuzzboard#5] [General]
The original Stanford document that I followed https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8055937/pdf/fphys-12-624317.pdf shows a pulse diagram that computes to 167ms pulses while the text speaks to 100ms.  A second document https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8771098/pdf/NRR-17-1495.pdf speaks to a random pattern for the finger stimulation (as opposed to the specific patterns in the first paper).

The standard board and glove software all follow pattern A from the first document and have a 100ms pulse width per the text.  I recently had a reach out from an individual that properly made a board, but was not getting a result.  I provided an updated motorshield sketch (only change was timer from 100 to 167) and they are getting a positive result.  I have added VibroTherayMotorShield_167.ino to the Google drive and the Github for general use.  Full disclosure, my first attempts built off of the 167ms timer as that was my initial math.

I am happy to modify the Sketches for pattern B as well as random if there is interest.

I would welcome a discussion on jitter and its potential value.

**Comment by PWPInnovator898** [comment]
Ideally, 167ms is the period (which is a 6 Hz frequency) between finger stimulation bursts. Within this period are 25 x 250 Hz pulses (=100ms).  So, there is 67 ms of the the period which are unused (quiet before the next burst).

**Reply by rebecca-sich** [reply]
So have you added a 67ms rest after each one or not really?

**Comment by garckenn** [comment]
Just finished building a set of gloves (using 3-finger archery gloves) for a friend.

I used the sketch called [VibroTherapyMotorShieldv3.ino](https://github.com/mchusma/vCR-reproduction/blob/main/VibroTherapyMotorShieldv3.ino) from @mchusma /vCR-reproduction.
- _"This was an attempt to recreate how the paper described the pattern, without recreating the diagram. It has much more randomness than the other projects."_

Is anyone familiar with it and understand how it compares to @pdbuzzboard 's original pattern, [VibroTherapyMotorShield](https://github.com/pdbuzzboard/pdbuzzboard/blob/main/VibroTherapyMotorShield.ino)?

Is it more or less Pattern A vs. Pattern B?

![Gloves Pic 1](https://github.com/user-attachments/assets/da2a996a-1745-46c6-9aa9-7fd72cbb9271)

![Gloves Pic 2](https://github.com/user-attachments/assets/5d01a704-7da0-4e86-b109-c3b39f6122a7)

**Comment by garckenn** [comment]
Here's what ChatGPT had to say. Seems logical

https://www.frontiersin.org/journals/physiology/articles/10.3389/fphys.2021.624317/full
https://pmc.ncbi.nlm.nih.gov/articles/PMC5836884/

![image](https://github.com/user-attachments/assets/8148a4d3-f0ac-4017-9e91-acaab322e15e)

Matching Patterns to Studies
**Study 1 (Frontiers in Physiology):**
- Likely involves specific frequency and amplitude patterns aimed at reducing tremor or improving motor function.
- The advanced customization and control of VibroTherapyMotorShieldv3.ino would better replicate the precise therapeutic effects described.

 **Study 2 (PMC Article):**
- Focuses on patterned tactile stimulation for neuromodulation.
- The ability to create consistent and well-timed patterns in V3 makes it a closer match for replicating results.

**Recommendation**
- Use VibroTherapyMotorShieldv3.ino as the base for your project. Its enhanced capabilities for pattern generation and motor control align better with the needs of Parkinson’s therapy as described in both studies.
- Consider examining the exact frequency and waveform settings used in the studies and ensure the selected code can replicate these parameters. If needed, modify the code to fine-tune frequency, amplitude, and timing based on the therapeutic protocols.
